packages/x-data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -1283,8 +1290,8 @@ describe('<DataGrid /> - Layout & warnings', () => {
         </div>,
       );
     }).toErrorDev([
-      'Warning: Encountered two children with the same key, `id`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.',
-      'Warning: Encountered two children with the same key, `id`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.',
+      'Encountered two children with the same key, `id`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.',
```

Comment:
How does this change work in both React 18 and React 19? 🤔

Reply:
console.error with R18
`"Warning: Encountered two children with the same key, `id`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at GridColumnHeaders (webpack-internal:///./packages/x-data-grid/src/components/columnHeaders/GridBaseColumnHeaders.tsx:39:7)
	    at GridColumnHeaders (webpack-internal:///./packages/x-data-grid/src/components/GridColumnHeaders.tsx:19:7)
	    at GridHeaders (webpack-internal:///./packages/x-data-grid/src/components/GridHeaders.tsx:21:73)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at GridTopContainer
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at eval (webpack-internal:///./packages/x-data-grid/src/components/virtualization/GridMainContainer.tsx:31:60)
	    at GridVirtualScroller (webpack-internal:///./packages/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx:62:59)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at GridRoot (webpack-internal:///./packages/x-data-grid/src/components/containers/GridRoot.tsx:34:60)
	    at GridContextProvider (webpack-internal:///./packages/x-data-grid/src/context/GridContextProvider.tsx:15:3)
	    at DataGrid (webpack-internal:///./packages/x-data-grid/src/DataGrid/DataGrid.tsx:33:56)
	    at div
	    at Wrapper (webpack-internal:///./node_modules/.pnpm/@mui+internal-test-utils@1.0.21_@babel+core@7.26.0_@types+react-dom@18.3.1_@types+react@18.3._dt4u4va7ipu67pjdrjfv5uv6ei/node_modules/@mui/internal-test-utils/build/createRenderer.js:430:35)"`
and
`
"Warning: Encountered two children with the same key, `id`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
	    at div
	    at GridRow (webpack-internal:///./packages/x-data-grid/src/components/GridRow.tsx:38:7)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at GridVirtualScrollerRenderZone (webpack-internal:///./packages/x-data-grid/src/components/virtualization/GridVirtualScrollerRenderZone.tsx:43:7)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at GridVirtualScrollerContent (webpack-internal:///./packages/x-data-grid/src/components/virtualization/GridVirtualScrollerContent.tsx:31:60)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at eval (webpack-internal:///./packages/x-data-grid/src/components/virtualization/GridMainContainer.tsx:31:60)
	    at GridVirtualScroller (webpack-internal:///./packages/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx:62:59)
	    at div
	    at eval (webpack-internal:///./node_modules/.pnpm/@emotion+react@11.13.3_@types+react@18.3.12_react@18.3.1/node_modules/@emotion/react/dist/emotion-element-7a1343fa.browser.development.esm.js:70:66)
	    at GridRoot (webpack-internal:///./packages/x-data-grid/src/components/containers/GridRoot.tsx:34:60)
	    at GridContextProvider (webpack-internal:///./packages/x-data-grid/src/context/GridContextProvider.tsx:15:3)
	    at DataGrid (webpack-internal:///./packages/x-data-grid/src/DataGrid/DataGrid.tsx:33:56)
	    at div
	    at Wrapper (webpack-internal:///./node_modules/.pnpm/@mui+internal-test-utils@1.0.21_@babel+core@7.26.0_@types+react-dom@18.3.1_@types+react@18.3._dt4u4va7ipu67pjdrjfv5uv6ei/node_modules/@mui/internal-test-utils/build/createRenderer.js:430:35)"
`

console.error with R19
`"Encountered two children with the same key, `id`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version."` x2

React 19 version is a part of the earlier version, so it works for both now